src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_CalendarView.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Private.Infrastructure;
+using Windows.UI.Xaml.Controls;
+using MUXControlsTestApp.Utilities;
+using Windows.UI.Xaml.Media;
+using Uno.UI.RuntimeTests.MUX.Helpers;
+using Windows.UI;
+using Windows.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Controls;
+using System.Reflection;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls;
+
+[TestClass]
+[RunsOnUIThread]
+public class Given_CalendarView
+{
+
+	[TestMethod]
+	public async Task SelectedDatesBorder()
+	{
+		DateTimeOffset day1 = new DateTimeOffset(DateTime.Now.AddDays(-3));
+		DateTimeOffset day2 = new DateTimeOffset(DateTime.Now.AddDays(4));
+		Type type = typeof(CalendarViewBaseItem);
```

Comment:
Since this doesn't build on Windows, the correct directive should be `#if !WINDOWS_UWP`

Reply:
Can I use below code instead of `#if !WINDOWS_UWP`
```
#if WINDOWS_UWP
[Ignore("Test doesn't buid on Windows")]
#endif
```

Reply:
No, because `typeof(CalendarViewBaseItem)` will still be compiled on Windows and will still fail.

Reply:
Hi, @Youssef1313 , @jeromelaban . Could you help me the azure pipeline of my pull request?

Reply:
```
1 unit test(s) failed (count=1).
	Failing Tests:
	1. SelectedDatesBorder()


---
	Details:
Failed: SelectedDatesBorder() [System.Reflection.TargetException] 
 System.Reflection.TargetException: Non-static method requires a target.
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00019] in &lt;44cb33854a8348c080262d2f16fbc92d&gt;:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in &lt;44cb33854a8348c080262d2f16fbc92d&gt;:0 
  at Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls.Given_CalendarView.SelectedDatesBorder () [0x001bd] in &lt;c4bc13a41fac4f2d84b3eb0290e08c40&gt;:0 
  at Uno.UI.Samples.Tests.UnitTestsControl+&lt;&gt;c__DisplayClass63_1.&lt;ExecuteTestsForInstance&gt;g__InvokeTestMethod|2 (Uno.UI.Samples.Tests.TestCase testCase) [0x0059b] in &lt;3c48dbda206d4d33843fa37902cdeab0&gt;:0 

```

The test appears to be passing on Skia and Wasm, but not Android/iOS/macOS.

I think `FindVisualChildrenByType` might be returning `null` on the failing platforms for some reason?

Reply:
https://github.com/unoplatform/uno/blob/64fb74e411e0d85e9311593a878ff1cf25055011/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TreeView.cs#L24-L29

https://github.com/unoplatform/uno/blob/64fb74e411e0d85e9311593a878ff1cf25055011/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TreeView.cs#L128-L130

https://github.com/unoplatform/uno/blob/64fb74e411e0d85e9311593a878ff1cf25055011/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TreeView.cs#L213-L217

I think above lines are related with this problem.
So I should add directives?
#if __MACOS__
[Ignore("Test is unstable on MacOS currently")]
#endif
#if __IOS__
[Ignore("Test is unstable on iOS currently")]
#endif
#if __ANDROID__
[Ignore("Test is unstable on Android currently")]
#endif

Reply:
Just

```csharp
#if __ANDROID__ || __IOS__ || __MACOS__
[Ignore("Test fails on these platforms")]
#endif
```

It would be best if you can understand "why" it fails on these platforms, but skipping should be fine IMO if you can't investigate this.